schemas/basicDataTypes/object/propertySpecifications.json

```diff
@@ -0,0 +1,15 @@
+{
+  "type": "object",
+  "propertyNames": {
+   "pattern": "[a-z]{5}"
+  },
+  "patternProperties": {
+    "^a": { "type": "string" },
+    "^b": { "type": "integer" }
+  },
+  "additionalProperties": {
+    "type": "boolean"
+  },
+  "minProperties": 1,
+  "maxProperties": 10

```

Comment:
I combined all the specifications of 
`propertyNames`, `patternProperties`, `additionalProperties` and size specs of `minProperties` & `maxProperties` 

all in ONE file since these would be listed together in the same section when displayed in the viewer.

P.S just in case anyone is curious, a valid JSON that passes this schema would be
```json
{
  "apple": "red fruit",
  "black": 255,
  "catty": true 
}
```